codec-http2/src/main/java/io/netty/handler/codec/http2/HttpToHttp2ConnectionHandler.java

```diff
@@ -116,4 +120,22 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
             promiseAggregator.doneAllocatingPromises();
         }
     }
+
+    private static void writeHeaders(ChannelHandlerContext ctx, Http2ConnectionEncoder encoder, int streamId,
+                                     HttpHeaders headers, Http2Headers http2Headers, boolean endStream,
+                                     SimpleChannelPromiseAggregator promiseAggregator) {
+        int dependencyId = headers.getInt(
+                HttpConversionUtil.ExtensionHeaderNames.STREAM_DEPENDENCY_ID.text(), -1);
```

Comment:
Should we just use `0` instead of `-1` and then just process everything the same way below?